RELEASE_PROCESS.md

```diff
@@ -127,10 +127,35 @@ based on the branch you just created.
 - [ ] Send an email to the chef and chef-dev mailing lists announcing
 the release with a link to the blog post.
 
-Chef Server is now released.
 
 ## Post Release
 
-- [ ] Bump the Chef Server version number in
-  `omnibus/config/projects/chef-server.rb` for development to the next
-  logical version number.
+Before the release can be considered complete, the following
+post-release steps must be performed;
+
+- [ ] in `omnibus/config/projects/chef-server.rb`:
+  - [ ] Bump the Chef Server version number for development to the next logical
+        version number.
+  - [ ] Update chef client to the latest stable chef-client version via
+        `override :chef`.  This may include updates to major and minor
```

Comment:
I'm not actually seeing chef locked atm. Is that coming in a separate PR @marcparadise?